### PR TITLE
start

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/mesh/core/solid_mesh.h [hunk 1]
```diff
@@ -274,14 +274,13 @@ namespace geode
          * Return the index in the mesh of a local vertex in a polyhedron.
          * @param[in] polyhedron_vertex Local index of vertex in polyhedron.
          */
-        index_t polyhedron_vertex(
-            const PolyhedronVertex& polyhedron_vertex ) const;
+        index_t vertex( const PolyhedronVertex& polyhedron_vertex ) const;
```
Comment (panquez) [inline include/geode/mesh/core/solid_mesh.h:277]: vertex or vertex_id ?

File: include/geode/mesh/core/solid_mesh.h [hunk 2]
```diff
@@ -298,49 +297,48 @@ namespace geode
          * @param[in] polyhedron_facet_vertex Local index of the vertex in the
          * facet of a polyhedron.
          */
-        index_t polyhedron_facet_vertex(
+        index_t vertex(
             const PolyhedronFacetVertex& polyhedron_facet_vertex ) const;
 
-        PolyhedronVertex polyhedron_facet_vertex_id(
+        PolyhedronVertex vertex_id(
```
Comment (panquez) [inline include/geode/mesh/core/solid_mesh.h:303]: what is a `vertex_id` and what is a `vertex`? I would swap the two method names

File: include/geode/mesh/core/surface_mesh.h [hunk 3]
```diff
@@ -182,62 +182,60 @@ namespace geode
          * Return the index in the mesh of a local vertex in a polygon
          * @param[in] polygon_vertex Local index of vertex in polygon
          */
-        index_t polygon_vertex( const PolygonVertex& polygon_vertex ) const;
+        index_t vertex( const PolygonVertex& polygon_vertex ) const;
```
Comment (panquez) [inline include/geode/mesh/core/surface_mesh.h:185]: vertex_id

File: include/geode/mesh/core/edged_curve.h [hunk 4]
```diff
@@ -54,9 +56,9 @@ namespace geode
 
         const Point< dimension >& point( index_t vertex_id ) const;
 
-        double edge_length( index_t edge_id ) const;
+        double length( index_t edge_id ) const;
```
Comment (BotellaA) [inline include/geode/mesh/core/edged_curve.h:59]: Not this one, we should revert this change

Comment (panquez) [reply include/geode/mesh/core/edged_curve.h:59]: why this one and not `barycenter`?

File: include/geode/mesh/core/graph.h [hunk 5]
```diff
@@ -99,6 +97,10 @@ namespace geode
         PASSKEY( GraphBuilder, GraphKey );
 
     public:
+        using Builder = GraphBuilder;
+        using EdgeVertices = std::array< index_t, 2 >;
```
Comment (panquez) [inline include/geode/mesh/core/graph.h:101]: Vertices? as done for SurfaceMesh and SolidMesh

File: include/geode/mesh/core/surface_mesh.h [hunk 6]
```diff
@@ -332,23 +331,15 @@ namespace geode
          */
         template < index_t T = dimension >
         typename std::enable_if< T == 3, absl::optional< Vector3D > >::type
-            polygon_vertex_normal( index_t vertex_id ) const;
+            vertex_normal( index_t vertex_id ) const;
 
         /*!
          * Get all the polygons with one of the vertices matching given vertex.
          * @param[in] vertex_id Index of the vertex.
```
Comment (panquez) [inline include/geode/mesh/core/surface_mesh.h:338]: maybe add a comment on the usage of the optional?

File: include/geode/mesh/core/solid_mesh.h [hunk 7]
```diff
@@ -293,54 +283,40 @@ namespace geode
...
-        index_t polyhedron_facet_vertex(
-            const PolyhedronFacetVertex& polyhedron_facet_vertex ) const;
-
-        PolyhedronVertex polyhedron_facet_vertex_id(
+        PolyhedronVertex polyhedron_vertex(
             const PolyhedronFacetVertex& polyhedron_facet_vertex ) const;
 
         /*!
          * Return the indices in the mesh of the two polyhedron edge vertices.
          * @param[in] polyhedron_facet_edge Local index of edge in a polyhedron.
```
Comment (panquez) [inline include/geode/mesh/core/solid_mesh.h:291]: Same remark as for Surface

File: include/geode/mesh/core/solid_mesh.h [hunk 8]
```diff
@@ -293,54 +283,40 @@ namespace geode
...
-        PolyhedronFacetVertices polyhedron_facet_vertices(
+        FacetVertices facet_vertices(
             const PolyhedronFacet& polyhedron_facet ) const;
 
-        virtual PolyhedronFacetsVertices polyhedron_facets_vertices(
-            index_t polyhedron ) const;
+        virtual FacetsVertices facets_vertices( index_t polyhedron ) const;
 
-        virtual PolyhedronFacets polyhedron_vertex_facets(
+        virtual PolyhedronFacets vertex_polyhedron_facets(
```
Comment (panquez) [inline include/geode/mesh/core/solid_mesh.h:311]: I do not understand that we get

Comment (panquez) [reply include/geode/mesh/core/solid_mesh.h:311]: is it all the PolyhedronFacets with this vertex? only in given polyhedron or all polyhedra or all polyhedra around vertex starting with given PolyhedronVertex?

File: include/geode/mesh/core/solid_mesh.h [hunk 9]
```diff
@@ -421,19 +396,11 @@ namespace geode
         /*!
          * Get all the polyhedra with one of the vertices matching given vertex.
          * @param[in] vertex_id Index of the vertex.
+         * @param[in] first_polyhedron One polyhedron index to begin research
```
Comment (panquez) [inline include/geode/mesh/core/solid_mesh.h:399]:
```suggestion
         * @param[in] first_polyhedron One polyhedron index to begin research (optional)

```

File: include/geode/mesh/core/solid_mesh.h [hunk 10]
```diff
@@ -476,29 +427,21 @@ namespace geode
...
-         * @param[in] vertices Indices of edge vertices.
-         * @pre This function needs that polyhedron adjacencies are computed
-         */
-        absl::optional< index_t > polyhedron_around_edge(
-            const std::array< index_t, 2 >& vertices ) const;
-
         /*!
          * Get all the polyhedra with both edge vertices.
          * @param[in] vertices Indices of edge vertices.
          * @param[in] first_polyhedron One polyhedron index to begin research.
```
Comment (panquez) [inline include/geode/mesh/core/solid_mesh.h:433]:
```suggestion
         * @param[in] first_polyhedron One polyhedron index to begin research (optional).

```

File: include/geode/mesh/core/solid_mesh.h (hunk 8 above)
Comment (BotellaA) [reply include/geode/mesh/core/solid_mesh.h:311]: PolyhedronFacets incident_polyhedron_facets_in_polyhedron(
            index_t polyhedron_id, index_t vertex_id ) const;

better?

File: include/geode/mesh/core/solid_mesh.h [hunk 11]
```diff
@@ -230,6 +210,17 @@ namespace geode
 
     public:
         using Builder = SolidMeshBuilder< dimension >;
+        using Vertices = absl::InlinedVector< index_t, 4 >;
+        using EdgeVertices = std::array< index_t, 2 >;
+        using EdgesVertices = absl::InlinedVector< EdgeVertices, 6 >;
+        using FacetVertices = absl::InlinedVector< index_t, 3 >;
+        using FacetsVertices = absl::InlinedVector< FacetVertices, 4 >;
+        using PolyhedronFacets = absl::InlinedVector< PolyhedronFacet, 4 >;
+        using PolyhedronFacetsOnBorder = PolyhedronFacets;
```
Comment (MelchiorSchuh) [inline include/geode/mesh/core/solid_mesh.h:219]: Est-ce qu'on ne mettrais pas un InlinedVector plus petit pour les PolyhedronFacetsOnBorder ?

File: include/geode/mesh/core/solid_mesh.h [hunk 12]
```diff
@@ -274,14 +265,13 @@ namespace geode
...
-        index_t polyhedron_vertex(
-            const PolyhedronVertex& polyhedron_vertex ) const;
+        index_t vertex( const PolyhedronVertex& polyhedron_vertex ) const;
 
         /*!
          * Return all the indices in the mesh of polyhedron vertices.
          * @param[in] polyhedron_id Index of polyhedron.
          */
-        PolyhedronVertices polyhedron_vertices( index_t polyhedron_id ) const;
+        Vertices polyhedron_vertices( index_t polyhedron_id ) const;
```
Comment (MelchiorSchuh) [inline include/geode/mesh/core/solid_mesh.h:274]: Didn't we say 'vertices' here ?

File: include/geode/mesh/core/solid_mesh.h [hunk 13]
```diff
@@ -293,54 +283,40 @@ namespace geode
...
-        PolyhedronVertex polyhedron_facet_vertex_id(
+        PolyhedronVertex polyhedron_vertex(
             const PolyhedronFacetVertex& polyhedron_facet_vertex ) const;
 
         /*!
          * Return the indices in the mesh of the two polyhedron edge vertices.
          * @param[in] polyhedron_facet_edge Local index of edge in a polyhedron.
          */
-        std::array< index_t, 2 > polyhedron_facet_edge_vertices(
+        EdgeVertices polyhedron_facet_edge_vertices(
```
Comment (MelchiorSchuh) [inline include/geode/mesh/core/solid_mesh.h:293]: idem here, 'edge_vertices' ?

File: include/geode/mesh/core/surface_mesh.h [hunk 14]
```diff
@@ -145,6 +137,11 @@ namespace geode
 
     public:
         using Builder = SurfaceMeshBuilder< dimension >;
+        using Vertices = absl::InlinedVector< index_t, 3 >;
+        using EdgeVertices = std::array< index_t, 2 >;
+        using PolygonEdgesOnBorder = absl::InlinedVector< PolygonEdge, 3 >;
```
Comment (MelchiorSchuh) [inline include/geode/mesh/core/surface_mesh.h:142]: idem as for solid, do we expect there to be 3 or that in most cases it'll be 0, 1 or at most 2 ?

File: include/geode/mesh/core/solid_mesh.h (hunk 8 above)
Comment (panquez) [reply include/geode/mesh/core/solid_mesh.h]: yes

**Review by panquez — APPROVED** [review]
